### DISCUSSION TITLE
[lspconfig] Cannot access configuration for setup

### DISCUSSION THREAD

**Discussion by fmorroni** [nvim-java/nvim-java#296] [Q&A]
Hi, I'm having trouble getting this plugin to work. I followed the installation instructions for LazyVim but I'm getting this errors:
```
[lspconfig] Cannot access configuration for setup. Ensure this server is listed in `server_configurations.md` or added as a custom server.
Failed to run `config` for nvim-lspconfig
...local/share/nvim/lazy/lazy.nvim/lua/lazy/core/loader.lua:383: attempt to call field 'setup' (a table value)
# stacktrace:
  - ~/.config/nvim/lua/user/config/lazy.lua:27
  - ~/.config/nvim/init.lua:2
```
[Here's my config](https://github.com/fmorroni/dotfiles/tree/main/.config/nvim).

Not sure if it's a recent bug or if I'm missing something in the configuration. But I've been looking for issues with similar problems and haven't found much. I did see mention that I should have an up to date version of Lazy (which I have, `version 11.14.1`) and someone saying he had to run `:MasonUpdate` for the plugin to work, but that didn't help.

Also mason hasn't automatically installed anything, although that may be because it never got the chance due to the `lspconfig` error.

Edit: Also just to clarify, I'm not using LazyVim. I'm making a custom config so I don't have any defaults included from LazyVim. I did try to look into the LazyVim repo to check what I may be missing, but there was a lot of stuff and they're not even using `nvim-java`, they use `nvim-jdtls` directly, so I had no clue what could be needed.
And finally, [this guy's config](https://github.com/ShenFeng312/nvim-config) mentioned in [this issue](https://github.com/nvim-java/nvim-java/issues/240) [seems to work](https://github.com/nvim-java/nvim-java/issues/240#issuecomment-2216975266) and we have pretty much the same configs regarding `nvim-java`.

**Comment by fmorroni** [comment]
Okay it was a problem with my setup, although I'm still not sure why it was wrong. But anyways using this config seems to have made it to work:
```lua
return {
  "nvim-java/nvim-java",
  dependencies = { "neovim/nvim-lspconfig" },
  config = function()
    require("java").setup({
      jdk = {
        auto_install = false,
      },
    })
    require('lspconfig').jdtls.setup({
      servers = {
        jdtls = {
          settings = {
            java = {
              configuration = {
                runtimes = {
                  {
                    name = "JavaSE-17",
                    path = "/usr/lib/jvm/java-17-openjdk/",
                  },
                  {
                    name = "JavaSE-21",
                    path = "/usr/lib/jvm/java-21-openjdk/",
                    default = true,
                  }
                }
              }
            }
          }
        },
      },
    })
  end,
}
```

**Reply by candeemis** [reply]
This worked for me too. Thanks for sharing.

**Reply by fmorroni** [reply]
Hi great to hear, did you perhaps tried using it on multi-module maven projects? Because for me it works on single module projects, but breaks on multi-module ones when I try to import from an external module.
